coderd/database/migrations/000301_system_user.up.sql

```diff
@@ -0,0 +1,51 @@
+ALTER TABLE users
+	ADD COLUMN is_system bool DEFAULT false NOT NULL;
+
+CREATE INDEX user_is_system_idx ON users USING btree (is_system);
+
+COMMENT ON COLUMN users.is_system IS 'Determines if a user is a system user, and therefore cannot login or perform normal actions';
+
+-- TODO: tried using "none" for login type, but the migration produced this error: 'unsafe use of new value "none" of enum type login_type'
+--  	 -> not sure why though? it exists on the login_type enum.
+INSERT INTO users (id, email, username, name, created_at, updated_at, status, rbac_roles, hashed_password, is_system, login_type)
+VALUES ('c42fdf75-3097-471c-8c33-fb52454d81c0', 'prebuilds@system', 'prebuilds', 'Prebuilds Owner', now(), now(),
+		'active', '{}', 'none', true, 'password'::login_type);
```

Comment:
Use login_type none to be even more explicit that no one should be able to auth into the account.

```suggestion
		'active', '{}', 'none', true, 'none'::login_type);
```

Reply:
I guess you missed the comment above 😆

Reply:
I changed this and tested. it worked. perhaps there was something else at play when this was written @dannykopping?

Reply:
```
coder@nowjosias:~/coder$ make -B gen/db
generate
ok      github.com/coder/coder/v2/coderd/database/gentest       0.017s
```

More verbose output for `make gen` but it doesn't fail.

Reply:
Hhmm, might be something about my environment then.
Great! Thanks for checking.

Reply:
OK, upon further investigation, the change made by @SasSwart was incorrect.

The error we're encountering is: `unsafe use of new value "none" of enum type login_type`.
The operative term here is "new value".

This relates to a warning (and ironically referring to the same type):

https://github.com/coder/coder/blob/d8e95001e878a7b3e9b766ce25926e98bd01aa9f/coderd/database/migrations/create_migration.sh#L12-L36

The `login_type` type was [originally](https://github.com/coder/coder/blob/d8e95001e878a7b3e9b766ce25926e98bd01aa9f/coderd/database/migrations/000001_base.up.sql#L7-L10) setup with just `password` and `github` as enum values.

The type has changed subsequently, but because we run all migrations in a single transaction, we cannot reference a new enum value of this type until it's committed.

Unsure how to proceed with this.

Reply:
Was there a migration that didn't create and rename that type?

Reply:
Yes, 2 of them:

`coderd/database/migrations/000126_login_type_none.up.sql` and `coderd/database/migrations/000195_oauth2_provider_codes.up.sql`

Both of those came after the admonition in `coderd/database/migrations/create_migration.sh` was added, which makes me think we need to add a linter to catch this.

Reply:
Yep, Migration № 126. I _think_ the right move is to retroactively re-write that migration to have the same effect but compatible with the transation scheme.

Reply:
Cool, that's a good idea 👍 thanks!